relay-quotas/src/redis.rs

```diff
@@ -383,6 +385,73 @@ mod tests {
         );
     }
 
+    /// Tests that a quota with and without namespace are counted separately.
+    #[test]
+    fn test_non_global_namespace_quota() {
+        let get_quota = |namespace: Option<MetricNamespace>| -> Quota {
+            Quota {
+                id: Some(format!("test_simple_quota_{}", uuid::Uuid::new_v4())),
+                categories: DataCategories::new(),
+                scope: QuotaScope::Organization,
+                scope_id: None,
+                limit: Some(5),
+                window: Some(60),
+                reason_code: Some(ReasonCode::new(format!("ns: {:?}", namespace))),
+                namespace,
+            }
+        };
+
+        let quotas = &[get_quota(None)];
+        let quota_with_namespace = &[get_quota(Some(MetricNamespace::Transactions))];
+
+        let scoping = ItemScoping {
+            category: DataCategory::Error,
+            scoping: &Scoping {
+                organization_id: 42,
+                project_id: ProjectId::new(43),
+                project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+                key_id: Some(44),
+            },
+            namespace: Some(MetricNamespace::Transactions),
+        };
+
+        let rate_limiter = build_rate_limiter();
+
+        for i in 0..10 {
+            let rate_limits: Vec<RateLimit> = rate_limiter
+                .is_rate_limited(quotas, scoping, 1, false)
+                .expect("rate limiting failed")
+                .into_iter()
+                .collect();
+
+            if i >= 5 {
+                assert_eq!(
+                    rate_limits[0].reason_code,
+                    Some(ReasonCode::new("ns: None"))
+                );
+            } else {
+                assert_eq!(rate_limits, vec![]);
+            }
+        }
+
+        for i in 0..10 {
```

Comment:
These two loops are asserting validating two different features. Could we split them into two separate tests, as per [Relay guidelines](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#fd14645580f848d9969d124a59cc3f7d) indicate?

> Keep unit tests constrained to a single concern. Do not test multiple things in one test, do not test sequences of actions. Write multiple tests for each of the concerns.

This makes identifying the root cause easier on test failures.

Reply:
so I don't agree here, the two loops are there two assert that with and without namespace is counted differently, this is only possible if we have them both in the same test to ensure theyre both connected to the same redis instance. if you look below this test at `fn test_simple_quota` then you see the original test that checked the behaviour without namespace